File: apina-core/src/test/java/fi/evident/apina/java/model/type/JavaBasicTypeTest.java
package fi.evident.apina.java.model.type;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertThat;

public class JavaBasicTypeTest {

    @Test
    public void primitiveNumbers() {
        assertThat(new JavaBasicType(int.class).isPrimitiveNumber(), is(true));
        assertThat(new JavaBasicType(long.class).isPrimitiveNumber(), is(true));
        assertThat(new JavaBasicType(short.class).isPrimitiveNumber(), is(true));
        assertThat(new JavaBasicType(float.class).isPrimitiveNumber(), is(true));
        assertThat(new JavaBasicType(double.class).isPrimitiveNumber(), is(true));

        assertThat(new JavaBasicType(boolean.class).isPrimitiveNumber(), is(false));
        assertThat(new JavaBasicType(char.class).isPrimitiveNumber(), is(false));
        assertThat(new JavaBasicType(Integer.class).isPrimitiveNumber(), is(false));
        assertThat(new JavaBasicType(String.class).isPrimitiveNumber(), is(false));
    }

    @Test
    public void wideTypes() {
        assertEquals(false, new JavaBasicType(int.class).isWide());
        assertEquals(false, new JavaBasicType(float.class).isWide());
        assertEquals(true, new JavaBasicType(long.class).isWide());
        assertEquals(true, new JavaBasicType(double.class).isWide());
    }
}

File: apina-core/src/test/kotlin/fi/evident/apina/java/model/type/JavaBasicTypeTest.kt
package fi.evident.apina.java.model.type

import org.junit.jupiter.api.Test
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class JavaBasicTypeTest {

    @Test
    fun primitiveNumbers() {
        assertTrue(JavaType.Basic(Int::class.javaPrimitiveType!!).isPrimitiveNumber)
        assertTrue(JavaType.Basic(Long::class.javaPrimitiveType!!).isPrimitiveNumber)
        assertTrue(JavaType.Basic(Short::class.javaPrimitiveType!!).isPrimitiveNumber)
        assertTrue(JavaType.Basic(Float::class.javaPrimitiveType!!).isPrimitiveNumber)
        assertTrue(JavaType.Basic(Double::class.javaPrimitiveType!!).isPrimitiveNumber)

        assertFalse(JavaType.Basic(Boolean::class.javaPrimitiveType!!).isPrimitiveNumber)
        assertFalse(JavaType.Basic(Char::class.javaPrimitiveType!!).isPrimitiveNumber)
        assertFalse(JavaType.Basic(Int::class.javaObjectType).isPrimitiveNumber)
        assertFalse(JavaType.Basic(String::class.java).isPrimitiveNumber)
    }

    @Test
    fun wideTypes() {
        assertFalse(JavaType.Basic(Int::class.javaPrimitiveType!!).isWide)
        assertFalse(JavaType.Basic(Float::class.javaPrimitiveType!!).isWide)
        assertTrue(JavaType.Basic(Long::class.javaPrimitiveType!!).isWide)
        assertTrue(JavaType.Basic(Double::class.javaPrimitiveType!!).isWide)
    }
}
